### PR TITLE
Add UpdateInstancesMetadata step

### PR DESCRIPTION
**Pull request by palaviv**
You can now update an instance metadata as one of the steps in the following way:
```json
"UpdateInstancesMetadata": [
          {
                "Instance": "test-daisy-instance",
                "Metadata": {
                        "test": "testing",
                        "test2": "testing2"
                }
          }
    ]
```

### CODE REVIEW

File: daisy/step_update_instances_metadata.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
+//  Copyright 2017 Google Inc. All Rights Reserved.
```
Comment (zoran15) [inline daisy/step_update_instances_metadata.go]: 2019

File: daisy/step_update_instances_metadata.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	// Metadata
+	Metadata map[string]string `json:"metadata,omitempty"`
+
+	// Instance to attach to.
+	Instance      string
+	project, zone string
+}
+
+func (c *UpdateInstancesMetadata) populate(ctx context.Context, s *Step) DError {
+	// We need to validate for duplicates agains the existing metadata so we will do this in the running.
```
Comment (zoran15) [inline daisy/step_update_instances_metadata.go]: typo: against

File: daisy/step_update_instances_metadata.go (hunk 1 above)
Comment (palaviv) [reply daisy/step_update_instances_metadata.go]: Fixed

File: daisy/step_update_instances_metadata.go (hunk 2 above)
Comment (palaviv) [reply daisy/step_update_instances_metadata.go]: Fixed

**Pull request merged**